packages/eslint-plugin/src/rules/prefer-find.ts

```diff
@@ -41,20 +41,41 @@ export default createRule({
       filterNode: TSESTree.Node;
     }
 
-    function parseIfArrayFilterExpression(
+    function parseArrayFilterExpressions(
       expression: TSESTree.Expression,
-    ): FilterExpressionData | undefined {
+    ): FilterExpressionData[] {
       if (expression.type === AST_NODE_TYPES.SequenceExpression) {
         // Only the last expression in (a, b, [1, 2, 3].filter(condition))[0] matters
         const lastExpression = nullThrows(
           expression.expressions.at(-1),
           'Expected to have more than zero expressions in a sequence expression',
         );
-        return parseIfArrayFilterExpression(lastExpression);
+        return parseArrayFilterExpressions(lastExpression);
       }
 
       if (expression.type === AST_NODE_TYPES.ChainExpression) {
-        return parseIfArrayFilterExpression(expression.expression);
+        return parseArrayFilterExpressions(expression.expression);
+      }
+
+      // This is the only reason we're returning a list rather than a single value.
```

Comment:
[Praise] I suspect from the code + comments + general flow of this rule that this was probably annoying to implement on your end. Thanks for going through that annoyance. 😄 

(I would be annoyed at it)

Reply:
haha actually i was pleasantly surprised to find that it was quite straightforward :) I was scared there would be complications with overlapping fixes or something, but everything just worked.